### PR TITLE
Use python 3.11 for peribolos check

### PR DESCRIPTION
**Pull request by dsimansk**
Related to https://github.com/knative/infra/issues/530

/cc @skonto @ReToCode @upodroid

### CODE REVIEW

File: peribolos/ci [hunk 1]
```diff
@@ -0,0 +1 @@
+Thu Oct 10 15:01:07 CEST 2024
```
Comment (dsimansk) [inline peribolos/ci:1]: Forcing an action run, should be removed before merge.

/hold

File: .github/workflows/peribolos-schema-validate.yaml [hunk 2]
```diff
@@ -29,7 +29,10 @@ jobs:
       with:
         fetch-depth: 0
         lfs: true
-    - run: pip install strictyaml --break-system-packages
+    - uses: actions/setup-python@v5
+      with:
+        python-version: "3.11"    
+    - run: pip install strictyaml
```
Comment (dsimansk) [inline .github/workflows/peribolos-schema-validate.yaml:35]: It seems we can go with `--break-system-packages` or pin Python to a version that works without the flag. I don't know if there are any drawbacks with `--break-system-packages`, but it's one liner in a check we always can revisit.

Comment (dsimansk) [reply .github/workflows/peribolos-schema-validate.yaml:35]: `pipx` doesn't work for libraries:
```
➜  community git:(pr/pipx-fix) pipx install strictyaml

No apps associated with package strictyaml or its dependencies. If you are attempting to install a library, pipx should not be used. Consider using pip or a similar tool instead.
```

File: peribolos/ci [hunk 3]
```diff
@@ -0,0 +1 @@
+Thu Oct 10 15:10:16 CEST 2024
```
Comment (dsimansk) [inline peribolos/ci:1]: Forcing an action run, should be removed in final version.

**Comment by ReToCode** [comment]
@dsimansk we went with the other approach, right?

**Review by ReToCode — APPROVED** [review]
/lgtm
/approve

**Pull request merged**